src/components/z-breadcrumb/index.tsx

```diff
@@ -0,0 +1,424 @@
+import {Component, Prop, h, State, Host, Listen, Element, Event, Watch} from "@stencil/core";
+import {
+  BreadcrumbHomepageVariant,
+  BreadcrumbPath,
+  BreadcrumbPathStyle,
+  KeyboardCode,
+  ListDividerType,
+  ListSize,
+  PopoverPosition,
+} from "../../beans";
+import {mobileBreakpoint} from "../../constants/breakpoints";
+import {EventEmitter} from "puppeteer";
+import {handleKeyboardSubmit} from "../../utils/utils";
+
+@Component({
+  tag: "z-breadcrumb",
+  styleUrl: "styles.css",
+  shadow: true,
+})
+export class ZBreadcrumb {
+  /* Accessibility references */
+  /* Overflow-menu: https://www.w3.org/WAI/ARIA/apg/patterns/menu-button/ */
+  /* Breadcrumb: https://www.w3.org/WAI/ARIA/apg/patterns/breadcrumb/ */
+  /* Focus on multiline link: https://www.w3.org/WAI/WCAG22/Understanding/focus-appearance.html */
+
+  @Element() hostElement: HTMLZBreadcrumbElement;
+
+  /** [optional] Path elements */
+  @Prop()
+  paths?: BreadcrumbPath[] | string;
+
+  /** [optional] Sets the path style */
+  @Prop({reflect: true})
+  pathStyle?: BreadcrumbPathStyle = BreadcrumbPathStyle.UNDERLINED;
+
+  /** [optional] Variant of first node */
+  @Prop()
+  homepageVariant?: BreadcrumbHomepageVariant = BreadcrumbHomepageVariant.ICON;
+
+  /** [optional] Sets max number of element to show */
+  @Prop()
+  maxNodesToShow? = 5;
+
+  /** [optional] Controls the behaviour on <a> tag click/enter */
+  @Prop()
+  preventFollowUrl? = false;
+
+  /** [optional] Sets max number of row for each path inside the popover. Zero equals unlimited */
+  @Prop()
+  overflowMenuItemRows? = 0;
+
+  /** [optional] Sets the maximun number of chars per single node */
+  @Prop()
+  truncateChar? = 30;
+
+  /** Handle mobile */
+  @State()
+  isMobile: boolean;
+
+  /** Detect whether the length of the nodes shown exceeds the container length */
+  @State()
+  hasOverflow = false;
+
+  @State()
+  popoverEllipsisOpen = false;
+
+  /** Emitted when preventFollowUrl=true to handle custom page transition */
+  @Event()
+  clickOnNode: EventEmitter;
+
+  @Listen("resize", {target: "window"})
+  handleResize(): void {
+    this.isMobile = window.innerWidth <= mobileBreakpoint;
+    if (this.wrapElement && this.wrapElement.scrollWidth > this.wrapElement.clientWidth) {
```

Comment:
Si è capito perché dava l'errore su `wrapElement`? Quand'è che diventa "non undefined"?

Reply:
Credo di essere arrivato a questa conclusione (ho messo dei console log in questo metodo): quando inizi a trascinare wrapElement è definito, infatti si entra nell'if perchè la troncatura viene fatta correttamente. Visto che il componente renderizza di nuovo per via del cambio di stato, ripassa da questo metodo (per wrapElement si vede un undefined dai console log, probabilmente il ref non è ancora definito in quel momento) e senza il controllo si spacca.